### ISSUE TITLE
Unable to lookup for metadata controlled by authority in edit view

### ISSUE THREAD

**Issue by PabloDeAlbu** [DSpace/dspace-angular#1758]
**Describe the bug**
When someone try to add a metadata controlled by authority from /items/[uuid]/edit/metadata, it's not possible to add the authority key. In DSpace 6 it is possible to do this through a "Lookup" button, as you can see in the following image taken from the demo

![1](https://user-images.githubusercontent.com/11927172/183926026-bfa0fa35-6d61-4526-8600-3b0b5ab8b601.jpg)

in dspace 7 you can only add a metadata, but not its authority key

![1](https://user-images.githubusercontent.com/11927172/183928552-e6d26567-8950-447a-99d9-459a91e0a511.jpg)

the way I have found to add the authority key is through the metadata-import feature, by specifying the authority key in a csv file 

**To Reproduce**
1. Configure a metadata to be controlled by authority, for example _dc.​contributor.​author_
2. try to add a metadata with an authority key from /items/[uuid]/edit/metadata

**Expected behavior**
There should be some lookup button like in dspace 6

**Comment by MarijkaA** [comment]
UNSW is certainly keen to see this issue addressed in a future DSpace version. This issue impacts a number of our archived items which require the addition and removal of authority controlled authors/creators. As recorded in this ticket, our only alternative to work-around this issue is to add 'name metadata' to our archived items. The addition of metadata only authors/creators to our archived items prevents our users from discovering other repository items from the same creator via 'Browse search' (functionality which utilises authority id).

**Comment by imilos** [comment]
+1 for this feature. There is a lookup functionality available in the review process, but no in **Edit item**. This is the only missing feature preventing update of my University's deployment to DSpace 7.x.

**Comment by kshepherd** [comment]
+1, indicating my interest for this feature too. I may have opportunity to spend some time on development

**Comment by hutattedonmyarm** [comment]
We've implemented this for our own needs at Universität Konstanz. I'm not sure if this is would be ready to issue a PR, but we're willing to share, so here is the relevant commit: https://github.com/hutattedonmyarm/dspace-angular/commit/db9d3d0f108180f85296b64ff31ecf2ad70e6bb8

A couple notes:
- We're still on DSpace 7.5, so this is based on 7.5. Some work might need to be done, to get it working on 7.6
- We do have some unusual fields which do *not* have a choices plugin configured, but are saving their regular value as authority value as well. We're handling these as well here, but that might not be of use to many others
- This solution requires checking config values from the backend, so make sure to add the relevant `choices.plugin` keys to your `rest.properties.exposed` configuration. If the "authority controlled but no choices plugin" variant is required as well, the relevant `authority.controlled` keys need to be added as well
    - To make things easier (and avoid having to specify every authority field in the config again) we patched `ConfigurationRestRepository` in the backend to allow wildcard values and specified `rest.properties.exposed = choices.plugin.*` in our configuration

I'm sure that someone with a deeper knowledge of Angular and or more motivation and time could come up with a better solution, but that's what we got

**Comment by alanorth** [comment]
Yes, this is a very useful tool to assist repository staff and prevent invalid metadata values.

**Comment by toniprieto** [comment]
Thanks for sharing @hutattedonmyarm 

Just before seeing your comment I was trying to solve this issue in a slightly different way. I've ended up implementing a solution that uses submit form components that implement value selectors for vocabularies controlled fields (`DsDynamicOneboxComponent` and `DsDynamicScrollableDropdownComponent`). This allows selecting values using the same fields used in submission form for hierarchical vocabularies, vocabularies defined with input-values ​​pairs in input-forms.xml (dc.type, dc.language.iso, ...) and vocabularies that work as an autocomplete field with authority keys.

It also use an existent api call to retrieve whether a metadata uses controlled vocabularies (see: [https://github.com/DSpace/RestContract/blob/main/vocabularies.md#search-methods)](https://github.com/DSpace/RestContract/blob/main/vocabularies.md#search-methods) ) and add a field to edit or remove a authority key.

I just submitted a draft PR to also sharing this implementation: https://github.com/DSpace/dspace-angular/pull/2653

It should be ready for testing to see if it may be a valid solution, although there is still work to do before it is fully completed (mainly adding test). I don't have much experience with Angular, there are probably some parts that can be done better. If I have time I will try to complete the PR. In any case, it would be great to have similar functionality in version 8.x.
